### PR TITLE
Major improvements to github based library syncs, removal of node-scheduler.

### PR DESCRIPTION
**Pull request by UnbounDev**
Primarily enhancements due to [this api issue](https://github.com/jsdelivr/api/issues/58).

- All libraries that are synced via their github repos now use a common (eTag cache enabled)
getFiles function.
- Node-scheduler actually recommends using setInterval if all you want to do is run a
task every other minute or so, so that is now what we do (spaced out by three minutes each).

### CODE REVIEW

File: lib/utils.js [hunk 1]
```diff
@@ -1,13 +1,127 @@
...
+  , mail = require("./mail");
+
+module.exports.contains = function contains(str) {
     return function(v) {
         return v.indexOf(str) >= 0;
     };
 };
 
-exports.startsWith = function startsWith(str) {
+module.exports.startsWith = function startsWith(str) {
```
Comment (megawac) [inline lib/utils.js]: Contains and startsWith are solved by the comatible lodash version fyi

File: serve.js [hunk 2]
```diff
@@ -110,38 +107,63 @@ function serve(config, cb) {
...
-      }
-    }.bind(null,name));
+        triggerTask(name);
+      }.bind(null,name),6*(i*1e4 + pattern.minute*1e4));
+    });
   });
 }
 
+function triggerTask(name,done) {
+
```
Comment (megawac) [inline serve.js]: Why aren't you using a promise pattern or alternate async pattern which would handle error handling in the RFC

File: tasks/bootstrap.js [hunk 3]
```diff
@@ -1,66 +1,91 @@
...
+  }
+
+  utils.githubGetFiles(github,repoOwner,repoName,filterFn,function(err,files) {
+
+    if(err) return cb(err);
+
+    var filtered = [];
+    _.each(files, function(file) {
+      if(_allowed(file)) {
+        filtered.push(file.path);
```
Comment (megawac) [inline tasks/bootstrap.js]: Why isn't this a `_.filter`

Comment (megawac) [inline tasks/bootstrap.js]: This is `var filtered = _.filter(_allowed)`

File: tasks/jsdelivr.js [hunk 4]
```diff
@@ -24,18 +24,33 @@ module.exports = function(_github) {
 
   return function(cb) {
 
-    getFiles(function(err,files,etags) {
+    var repoOwner = "jsdelivr"
+      , repoName = "jsdelivr"
+      , filterFn = function(v) {
+        return v.name === 'files';
+      };
```
Comment (megawac) [inline tasks/jsdelivr.js]: Nit: `_.matchesProperty('name', 'files')`

File: tasks/jsdelivr.js [hunk 5]
```diff
@@ -24,18 +24,33 @@ module.exports = function(_github) {
...
-          log.err("failed to save jsdelivr etags ",err);
-        } else {
-          log.info("jsdelivr etags saved to " + etagsFilePath);
-        }
+        return true
+      }
+
+      _.each(files, function(file) {
+        if(_allowed(file))
+          filtered.push(file.path);
```
Comment (megawac) [inline tasks/jsdelivr.js]: Another filter

File: lib/utils.js (hunk 1 above)
Comment (UnbounDev) [inline lib/utils.js]: Oh I know - that code is a relic of another author, I'm slowly eradicating its usage.

File: serve.js (hunk 2 above)
Comment (UnbounDev) [inline serve.js]: Morally opposed to promises (not totally), if we're going to accept promises as the standard coding style for the project I'll conform - though generally I prefer not to. I'm not using an async pattern as that function must be called from a scheduled job (on an interval).

File: tasks/bootstrap.js (hunk 3 above)
Comment (UnbounDev) [inline tasks/bootstrap.js]: `_.filter`: that's helpful - thanks.

**Pull request merged**